pkg/quota/admission/podlimitrequest/admission.go

```diff
@@ -0,0 +1,125 @@
+package podlimitrequest
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/openshift/origin/pkg/cmd/server/api"
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"speter.net/go/exp/math/dec/inf"
+)
+
+const (
+	annEnabled         = "openshift.io/PodLimitRequestEnabled"
+	cpuBaseScaleFactor = 1000.0 / (1024.0 * 1024.0 * 1024.0) // 1000 milliCores per 1GiB
+)
+
+func init() {
+	admission.RegisterPlugin("PodLimitRequest", func(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+		return newPodLimitRequest(client, config)
+	})
+}
+
+type podLimitRequest struct {
+	*admission.Handler
+	Config     api.PodLimitRequestConfig
+	KubeClient kclient.Interface
+}
+
+// newPodLimitRequest returns an admission controller for containers that
+// configurably overrides container resource request/limits
+func newPodLimitRequest(client kclient.Interface, config io.Reader) (admission.Interface, error) {
+	glog.V(5).Infof("PodLimitRequest admission controller is loaded")
+	parsed, err := ReadConfig(config)
+	glog.V(5).Infof("PodLimitRequest admission controller got config: %v\nerror: %v", parsed, err)
+	return &podLimitRequest{
+		Handler:    admission.NewHandler(admission.Create),
+		Config:     parsed,
+		KubeClient: client,
+	}, err
+}
+
+func ReadConfig(configFile io.Reader) (api.PodLimitRequestConfig, error) {
+	glog.V(5).Infof("PodLimitRequest about to read config:\n%v", configFile)
+	config := api.PodLimitRequestConfig{}
+	buffer := new(bytes.Buffer)
+	buffer.ReadFrom(configFile)
+	glog.V(5).Infof("PodLimitRequest read config:\n%v", buffer.Bytes())
+	err := yaml.Unmarshal(buffer.Bytes(), &config)
+	if err == nil && config.Enabled {
```

Comment:
Well, it would have to return `config, error`. But it already does that. I read this as "if all is well and this plugin applies, do the logic. Otherwise return any errors." Does this really not seem more elegant, or is the "err != nil" convention really the only way to write this?